core/primitives/src/types.rs

```diff
@@ -27,6 +27,23 @@ pub type Balance = u128;
 /// Gas is a type for storing amount of gas.
 pub type Gas = u64;
 
+/// Number (or amount) of blocks (or block heights) in current context.
+// TODO discuss
+// Technically, blocks size is not equal to heights difference.
+// For example, 50 blocks don't imply heights from H to (H + 50).
+// It may lead to incorrect assumptions in future.
+// We can replace NumBlocks with HeightHorizon or something better.
```

Comment:
I suppose `NumBlocks` and `HeightDelta` may work good enough.
For example, `epoch_length` means difference between heights when new epoch should start - use `HeightDelta`.
In initial state sync we want to request some number of blocks disregarding their heights - use `NumBlocks`.